internal/acceptance/service_principal_test.go

```diff
@@ -14,6 +21,76 @@ const awsSpn = `resource "databricks_service_principal" "this" {
 	display_name = "SPN {var.RANDOM}"
 }`
 
+func TestAccServicePrincipalHomeDeleteSuccess(t *testing.T) {
+	appId := "12345a67-8b9c-0d1e-23fa-4567b89cde04"
+	appId2 := "22345a67-8b9c-0d1e-23fa-4567b89cde04"
+	os.Setenv("appId", appId)
+	os.Setenv("appId2", appId2)
+	workspaceLevel(t, step{
+		Template: `
+			resource "databricks_service_principal" "a" {
+				application_id = "{env.appId}"
+				force_delete_home_dir = true
+			}`,
+		Check: func(s *terraform.State) error {
```

Comment:
you can iterate over the `*terraform.State` here and pick up the application ID and set it to a test-level variable, that you'll use in the check in the next step

Reply:
I think the new implementation is much cleaner and aligns with your comments. Lmk if there are issues